linux.h

```diff
@@ -0,0 +1,111 @@
+#include <string.h>
+#include <unistd.h>
+
+#include "types.h"
+
+static inline int
+thread_stats__linux (thread_stats_t *stats, size_t *len) {
+  int err;
+
+  int self = getpid();
+  clock_t clock_ticks = sysconf(_SC_CLK_TCK);
+
+  float uptime;
+
+  int uptime_fd;
+  uptime_fd = open("/proc/uptime", O_RDONLY);
+
+  if (uptime_fd == -1) {
+    return -1;
+  }
+
+  char uptime_buf[4096 + 1];
+  int uptime_buf_read;
+
+  uptime_buf_read = read(uptime_fd, uptime_buf, 4096);
+
+  if (uptime_buf_read == -1) {
+    return -1;
+  }
+
+  err = close(uptime_fd);
+
+  if (err == -1) {
+    return -1;
+  }
```

Comment:
```suggestion
  err = close(uptime_fd);
  assert(err == 0);
```

Closing valid file descriptors is one of those things that aren't allowed to fail in practice so it's safe to assert to it succeeds.